content/en/docs/concepts/services-networking/service.md

```diff
@@ -310,43 +384,53 @@ REDIS_MASTER_PORT_6379_TCP_PORT=6379
+cluster using an [add-on](/docs/concepts/cluster-administration/addons/).
+
+A cluster-aware DNS server, such as CoreDNS, watches the Kubernetes API for new
+Services and creates a set of DNS records for each one.  If DNS has been enabled
+throughout your cluster then all Pods should automatically be able to resolve
+Services by their DNS name.
 
-For example, if you have a `Service` called `"my-service"` in a Kubernetes
-`Namespace` called `"my-ns"`, a DNS record for `"my-service.my-ns"` is created.  `Pods`
-which exist in the `"my-ns"` `Namespace` should be able to find it by simply doing
-a name lookup for `"my-service"`.  `Pods` which exist in other `Namespaces` must
-qualify the name as `"my-service.my-ns"`.  The result of these name lookups is the
-cluster IP.
+For example, if you have a Service called `"my-service"` in a Kubernetes
+Namespace `"my-ns"`, the control plane and the DNS service acting together will
+create a DNS record for `"my-service.my-ns"`. Pods in the `"my-ns"` Namespace
+should be able to find it by simply doing a name lookup for `my-service`
+(`"my-service.my-ns"` would also work).
+
+Pods n other Namespaces must qualify the name as `my-service.my-ns`. These names
+will resolve to the cluster IP assigned for the Service.
 
 Kubernetes also supports DNS SRV (service) records for named ports.  If the
-`"my-service.my-ns"` `Service` has a port named `"http"` with protocol `TCP`, you
-can do a DNS SRV query for `"_http._tcp.my-service.my-ns"` to discover the port
-number for `"http"`.
+`"my-service.my-ns"` Service has a port named `"http"` with protocol set to
+`TCP`, you can do a DNS SRV query for `_http._tcp.my-service.my-ns` to discover
+the port number for `"http"`, as well as the IP address.
 
-The Kubernetes DNS server is the only way to access services of type
-`ExternalName`.  More information is available in the [DNS Pods and
-Services](/docs/concepts/services-networking/dns-pod-service/).
+The Kubernetes DNS server is the only way to access `ExternalName` Services.
+You can find more information about `ExternalName` resolution in
+[DNS Pods and Services](/docs/concepts/services-networking/dns-pod-service/).
 
 ## Headless services
 
 Sometimes you don't need or want load-balancing and a single service IP.  In
-this case, you can create "headless" services by specifying `"None"` for the
-cluster IP (`.spec.clusterIP`).
+this case, you can create what are termed “headless” Services, by explicitly
+specifying `"None"` for the cluster IP (`.spec.clusterIP`).
 
-This option allows developers to reduce coupling to the Kubernetes system by
-allowing them freedom to do discovery their own way.  Applications can still use
-a self-registration pattern and adapters for other discovery systems could easily
+You can use a headless Service to interface with other service discovery mechanisms,
+without being tied to Kubernetes' implementation. For example, you could implement
+a custom [Operator](
```

Comment:
This line seems to be unfinished. @sftim could you please verify?

Reply:
Now fixed.